packages/mobile/src/geth/saga.ts

```diff
@@ -98,6 +101,76 @@ export function* waitForGethConnectivity() {
   }
 }
 
+function* pollGethStatusUntilSynced() {
+  let head: BlockHeader = yield select(chainHeadSelector)
+
+  // Head may be null on startup
+  if (!head) {
+    head = (yield take(Actions.SET_CHAIN_HEAD)).head
+  }
+
+  while (true) {
+    if (blockIsFresh(head)) {
+      return true
+    }
+
+    head = (yield take(Actions.SET_CHAIN_HEAD)).head
+  }
+}
+
+export function* waitForGethSync() {
+  const fornoEnabled = yield select(fornoSelector)
+  if (fornoEnabled) {
+    return
+  }
+
+  const { result } = yield race({
```

Comment:
Why implement a timeout here instead of at the call site? I think there are tradeoffs to both, but I think that having a constant timeout here may lead to issues for cases where the caller is willing to wait for longer periods of time. Maybe a good compromise is to make the timeout configurable.

Reply:
That's an interesting point that I didn't consider. I was thinking about it in a binary way: either you need it to be synced or you don't. What use cases do you think would benefit from configurable timeout durations?